modules/openid_auth/app/controllers/openid_auth/v2/validation_controller.rb

```diff
@@ -6,14 +6,42 @@
 module OpenidAuth
   module V2
     class ValidationController < ApplicationController
-      before_action :validate_user
+      before_action :validate_user, :validate_strict_audience
 
       def index
         render json: validated_payload, serializer: OpenidAuth::ValidationSerializerV2
       rescue => e
         raise Common::Exceptions::InternalServerError, e
       end
 
+      def valid_strict?
+        %w[true false].include?(fetch_strict)
+      end
+
+      def fetch_strict
+        params['strict'] || 'false'
+      end
+
+      def valid_audience?
+        aud = fetch_aud
+        if fetch_strict == 'true'
+          if aud.nil?
+            false
+          else
+            aud.include?(token.payload['aud'])
+          end
+        elsif aud.nil?
+          token.payload['aud'] == Settings.oidc.isolated_audience.default
+        else
+          [Settings.oidc.isolated_audience.default, aud].include?(token.payload['aud'])
```

Comment:
This will need to be called before the `valid_token` method in token.rb is called for it to have its intended results.

Reply:
when aud was an array the `.include?` method was trying to process an array inside an array, which doesn't work. solution creates a one-level array including all aud values as well as the default aud value.